File: lib/models/bitcoin.address.model.dart
class BitcoinAddressModel {
  int? id;
  int walletID;
  int accountID;
  String bitcoinAddress;
  int bitcoinAddressIndex;
  int inEmailIntegrationPool;
  int used;
  String serverWalletID;
  String serverAccountID;

  BitcoinAddressModel({
    required this.id,
    required this.walletID,
    required this.accountID,
    required this.serverWalletID,
    required this.serverAccountID,
    required this.bitcoinAddress,
    required this.bitcoinAddressIndex,
    required this.inEmailIntegrationPool,
    required this.used,
  });

  Map<String, dynamic> toMap() {
    return {
      'id': id,
      'walletID': walletID,
      'accountID': accountID,
      'serverWalletID': serverWalletID,
      'serverAccountID': serverAccountID,
      'bitcoinAddress': bitcoinAddress,
      'bitcoinAddressIndex': bitcoinAddressIndex,
      'inEmailIntegrationPool': inEmailIntegrationPool,
      'used': used,
    };
  }

  factory BitcoinAddressModel.fromMap(Map<String, dynamic> map) {
    return BitcoinAddressModel(
      id: map['id'],
      walletID: map['walletID'],
      accountID: map['accountID'],
      serverWalletID: map['serverWalletID'],
      serverAccountID: map['serverAccountID'],
      bitcoinAddress: map['bitcoinAddress'],
      bitcoinAddressIndex: map['bitcoinAddressIndex'],
      inEmailIntegrationPool: map['inEmailIntegrationPool'],
      used: map['used'],
    );
  }
}

File: rust/src/proton_wallet/db/model/bitcoin_address_model.rs
use rusqlite::{Result, Row};
use serde::{Deserialize, Serialize};
use serde_rusqlite::from_row;

use super::model::ModelBase;

#[derive(Debug, Serialize, Deserialize, Clone)]
pub struct BitcoinAddressModel {
    pub id: Option<u32>,
    pub server_id: String,
    pub wallet_id: u32,
    pub account_id: u32,
    pub bitcoin_address: String,
    pub bitcoin_address_index: u32,
    pub in_email_integration_pool: u32,
    pub used: u32,
    pub server_wallet_id: String,
    pub server_account_id: String,
}

impl ModelBase for BitcoinAddressModel {
    fn from_row(row: &Row) -> Result<Self> {
        Ok(from_row::<BitcoinAddressModel>(row).unwrap())
    }
}
